### PR TITLE
Adding static publish/multicast variants, and deprecating operators that return ConnectableObservable

### PR DESCRIPTION
**Pull request by benlesh**
Work in progress....

### CODE REVIEW

**Comment by JWO719** [comment]
In general I like the idea of splitting that up due to it's differnet behavior. But one think that came to my mind, isn't it confusing for api users to have a static operator and a pipeable operator with the same name but a different behavior? I don't know if it would be a good idea to give one of those two a different name in that case.

**Comment by cartant** [comment]
In #3833, the operator names had a `With` suffix - e.g. `publishWith` - to distinguish them from the static creators. I quite liked that idea. Is there a reason for not going in that direction? Perhaps because the usage is a little different to the `concatWith` in #3798? Is that issue still a thing?

**Comment by benlesh** [comment]
>  Is there a reason for not going in that direction?

This is still a WIP, :), hence the label. I just didn't get to the aliasing/deprecations yet.

>  But one think that came to my mind, isn't it confusing for api users to have a static operator and a pipeable operator with the same name but a different behavior?

Perhaps... although right now we have one operator that behaves two different ways depending on how you call it.  I think once we have `publishWith` or `publishAs` or whatever we want to call it, it will make more sense.

**Comment by BioPhoton** [comment]
> we have one operator that behaves two different ways depending on how you call it

This is also the case with a selector parameter in `multicast` and `publish`. Here the output is completely different.
`multicast` itself should be at least 2 different operator as well as static methods I believe.

Also, shareReplay has completely different behavior if refCount true is used.

In my opinion, the whole multicasting section should be checked. There are a lot of historical edge cases etc included which could get cleaned up or deprecated.

File: src/internal/observable/publishReplay.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Observable } from '../Observable';
+import { ReplaySubject } from '../ReplaySubject';
+import { multicast } from './multicast';
+import { ConnectableObservable } from '../observable/ConnectableObservable';
+
+export function publishReplay<T>(
+  source: Observable<T>,
+  bufferSize = Number.POSITIVE_INFINITY,
```
Comment (LayZeeDK) [inline src/internal/observable/publishReplay.ts]: `Number.MAX_SAFE_INTEGER`?

File: src/internal/observable/publishReplay.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  windowTime = Number.POSITIVE_INFINITY
```
Comment (LayZeeDK) [inline src/internal/observable/publishReplay.ts]: `Number.MAX_SAFE_INTEGER`?

**Comment by benlesh** [comment]
Closing this older PR in favor of work (and renewed discussion) here: #5432

**Pull request closed without merging**